assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -361,21 +362,22 @@ const SelectedVehicleDataLayers = ({
               />
             </SelectionCardContainer>
           )}
-          {isVehicle(selectedVehicleOrGhost) &&
-            (selectedVehicleOrGhost?.isShuttle ? (
+          {isVehicle(selectedVehicleOrGhost) && (
+            <>
               <VehicleMarker
                 key={selectedVehicleOrGhost.id}
                 vehicle={selectedVehicleOrGhost}
                 isPrimary={true}
                 isSelected={true}
               />
-            ) : (
+
               <RouteVehicles
                 selectedVehicleRoute={selectedVehicleOrGhost.routeId}
                 selectedVehicleId={selectedVehicleOrGhost.id}
                 onPrimaryVehicleSelect={selectVehicle}
               />
-            ))}
+            </>
```

Comment:
question(non-blocking): Based on my understanding, if the selected vehicle changes route, this will automatically switch over to rendering the new route shape + secondary vehicles because `selectedVehicleRoute={selectedVehicleOrGhost.routeId}` will change. Is that correct? 

I'm trying to wrap my head around the bug video and am getting stuck on why the route shape didn't clear if the primary vehicle was no longer assigned to the 23. That may be separate from the focus of this ticket though!

Reply:
That seems correct based on my observations.

Honestly I'm still trying to wrap my head around the _exact_ set of circumstances needed to reproduce this as well. The common cases I run through all seem to be handled more-or-less correctly in Skate already, and it's hard to find an example of a bus that's going to go off-route. Maybe I could do some poking around with bus routes that have detour alerts up.

Reply:
Yeah it seems tricky to reproduce! An assumption I made in the comment above about the video was wrong having watched at it again 
> am getting stuck on why the route shape didn't clear if the primary vehicle was no longer assigned to the 23.
I think the vehicle actually was still assigned to the 23 since the VPC still showed 23. So My concern about the route changing was not relevant! 

In any case, maybe it would be worth adding a test case where the selected vehicle is *not* returned by `useVehiclesForRoute` but is returned by `useVehicleForId`?

Reply:
> In any case, maybe it would be worth adding a test case where the selected vehicle is not returned by `useVehiclesForRoute` but is returned by `useVehicleForId`?

Adding my support for adding this test case!